src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1523,10 +1523,39 @@ private IEnumerable<ITypeSymbol> InferTypeInYieldStatement(YieldStatementSyntax
 
             private IEnumerable<ITypeSymbol> InferTypeForReturnStatement(ReturnStatementSyntax returnStatement, SyntaxToken? previousToken = null)
             {
+                bool isAsync;
+                IEnumerable<ITypeSymbol> types;
+
+                InferTypeForReturnStatement(returnStatement, previousToken, out isAsync, out types);
+
+                if (!isAsync)
+                {
+                    return types;
+                }
+
+                var taskOfT = this.Compilation.TaskOfTType();
+                if (taskOfT == null || types == null)
+                {
+                    return SpecializedCollections.EmptyEnumerable<ITypeSymbol>();
```

Comment:
Yeah, they're treated equivalently.  I agree the inconsistency is not ideal.  Ideally we can do a sweep through this at some point to unify things.  But it's low on my list of priorities.